vpr/src/route/connection_box_lookahead_map.cpp

```diff
@@ -92,78 +120,65 @@ class CostMap {
         return cost_map_[from_seg_index][size_t(box_id)][dx][dy];
     }
 
-    void set_cost_map(int from_seg_index,
-                      const t_routing_cost_map& cost_map,
-                      e_representative_entry_method method) {
-        const auto& device_ctx = g_vpr_ctx.device();
-        for (size_t box_id = 0;
-             box_id < device_ctx.connection_boxes.num_connection_box_types();
-             ++box_id) {
-            set_cost_map(from_seg_index, ConnectionBoxId(box_id), cost_map, method);
-        }
-    }
-
-    void set_cost_map(int from_seg_index, ConnectionBoxId box_id, const t_routing_cost_map& cost_map, e_representative_entry_method method) {
-        VTR_ASSERT(from_seg_index >= 0 && from_seg_index < (ssize_t)offset_.size());
+    // Set cost map from sparse hash table of data.
+    void set_cost_map(const routing_data_t& data, e_representative_entry_method method) {
+        vtr::NdMatrix<t_bb, 2> bounds({cost_map_.dim_size(0), cost_map_.dim_size(1)},
+                                      {0, 0, 0, 0});
 
         // Find coordinate offset for this segment.
-        int min_dx = 0;
-        int min_dy = 0;
-        int max_dx = 0;
-        int max_dy = 0;
-        for (const auto& entry : cost_map) {
-            if (std::get<3>(entry) != box_id) {
-                continue;
-            }
-            min_dx = std::min(std::get<1>(entry).first, min_dx);
-            min_dy = std::min(std::get<1>(entry).second, min_dy);
-
-            max_dx = std::max(std::get<1>(entry).first, max_dx);
-            max_dy = std::max(std::get<1>(entry).second, max_dy);
-        }
+        for (const auto& entry : data) {
```

Comment:
Please define an operation to combine bounding boxes.

Reply:
But that's not what this is?  This creating a bounding box from points?